### PR TITLE
Use GItHub Package now for mobile-native-android

### PR DESCRIPTION
**Pull request by toddsouthenbentley**

### CODE REVIEW

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -27,35 +27,10 @@ jobs:
...
-      env:
-        GITHUB_TOKEN: ${{ secrets.GH_PAT }}
-        
     - name: Build mobile-sdk-android
       run: |
         cd mobile-sdk-android
         chmod +x gradlew
         ./gradlew detekt build
+      env:
+        TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (tcobbs-bentley) [inline .github/workflows/android.yml:36]: What about USERNAME?

Comment (tcobbs-bentley) [reply .github/workflows/android.yml:36]: Also, the old stuff used secrets.GH_PAT. How did that change to secrets.GITHUB_TOKEN?

Comment (toddsouthenbentley) [reply .github/workflows/android.yml:36]: GITHUB_TOKEN is the built in PAT which is automatically generated by GitHub. When using that PAT, I don't know what the user name is and it works without it as evidenced by the successful CI build on this PR's branch.

Comment (tcobbs-bentley) [reply .github/workflows/android.yml:36]: If that's true (which it seems to be), then I think we should change settings.gradle to either not set the username at all, or set it to "token" (which is what somebody else has done [here](https://github.com/komputing/KHash/blob/master/settings.gradle.kts).) That doesn't belong in this PR, though.

**Review by tcobbs-bentley — APPROVED** [review]

File: .github/workflows/android.yml (hunk 1 above)
Comment (toddsouthenbentley) [reply .github/workflows/android.yml:36]: Interesting link, digging deeper it looks like we could create a PAT that ONLY has read:packages scope and then encode it. Apparently if we publicize the encoded version, it won't get automatically deleted by GitHub. That might be something to consider down the road if we get pushback from our users.

Comment (toddsouthenbentley) [reply .github/workflows/android.yml:36]: We can do this for the USERNAME (if we want): `USERNAME: ${{ github.actor }}`
This matches [gradle-publish.yml](https://github.com/iTwin/mobile-native-android/blob/66c47843e57380470be4cb05961f52cb8332a6c2/.github/workflows/gradle-publish.yml#L39) in mobile-native-android.

**Pull request merged**